src/Components/SysTable/hooks.js

```diff
@@ -0,0 +1,135 @@
+import { useApolloClient } from '@apollo/client';
+import { useEffect, useLayoutEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+export const useInventoryUtilities = (
+    inventory
+) => {
+    const dispatch = useDispatch();
+
+    // Resets the Inventory to a loading state
+    // and prevents previously shown columns and rows to appear
+    useLayoutEffect(() => {
```

Comment:
Is the `useLayoutEffect` required here? I think it's enough to use just `useEffect` if that's the case, might be worth moving it to bottom with `resetPage` effect